pkg/build/api/v1beta1/types.go

```diff
@@ -111,6 +111,10 @@ type BuildSource struct {
 	// This allows to have buildable sources in directory other than root of
 	// repository.
 	ContextDir string `json:"contextDir,omitempty"`
+
+	// SourceSecretName is the name of a Secret that would be used for setting
+	// up the authentication for cloning private repository.
```

Comment:
i'm not looking for major detail, but "secret must be an SSH key for git auth" or something...  our API turns out to be our docs frequently.